### DISCUSSION TITLE
Cloudflare被墙或请求超时的临时解决方案

### DISCUSSION THREAD

**Discussion by xfgryujk** [xfgryujk/blivechat#187] [教程]
> v1.9.3编辑：以下内容适用于v1.9.3之前的版本，从v1.9.3开始换了新域名，而且应该基本不会遇到被墙的情况。当然你也可以用这里的方法修改新的域名在Cloudflare节点的IP，新域名的说明参考[#244](https://github.com/xfgryujk/blivechat/discussions/244)

⚠️命令行警告，看到命令行就头晕的人可以关掉了⚠️

如果浏览器打不开chat.bilisc.com，或者后台日志出现`TimeoutError`，可能是你到Cloudflare的线路太差，或者因为众所周知的原因部分地区就是连接不上。最好的解决方法是开梯子使用blivechat，如果你没有梯子也可以按照下面的方法，修改chat.bilisc.com的IP到任意一个能用的IP

1. 首先下载[CloudflareSpeedTest](https://github.com/XIU2/CloudflareSpeedTest?tab=readme-ov-file#-%E5%BF%AB%E9%80%9F%E4%BD%BF%E7%94%A8)，解压
2. 打开`CloudflareST.exe`所在目录，按住Shift键，点击鼠标右键，在菜单中点击`在此处打开 Powershell 窗口`
3. 首先要确认你没开代理，否则测出来的结果不准。在弹出的控制台中输入以下命令，然后按回车：

    ```bat
    .\CloudflareST.exe -httping -url https://chat.bilisc.com/
    ```

4. 等待测速结果
5. 按下Win键 + X，在菜单中点击`终端管理员`
6. 在弹出的控制台中输入以下命令，然后按回车：

    ```bat
    notepad C:\Windows\System32\drivers\etc\hosts
    ```

7. 在弹出的记事本中，在文件末尾加上这样一行：

    ```
    <新的IP> chat.bilisc.com
    ```

    这里`<新的IP>`要替换成测速结果中任意一个IP，比如：

    ```
    172.66.43.83 chat.bilisc.com
    ```

8. 保存之后，用浏览器打开 https://chat.bilisc.com/ ，按下Shift + F5，看看访问速度怎么样。如果还是连接不上或者超时，再换个IP

**Comment by 1349033911** [comment]
福建地区全部阵亡
